config: Add printer-robo3d-r2-v2.1.5.cfg printer config

jschuh opened the pull request:
Adds Robo3D R2 printer config (successfully tested).

Signed-off-by: Justin Schuh <code@justinschuh.com>

KevinOConnor commented:
Thanks.  In general, it looks fine to me.  Does the stock printer really have a filament sensor hooked up to a rpi?  The convention for files that start with "printer-" to have a year associated with it.  What year, roughly, is this printer from?

-Kevin

jschuh commented:
Yep, the stock printer really does have the filament runout wired to the rpi, even though there's no shortage of available pins on the board. I definitely was not expecting that when I started tracing the wires, but below is an image of what the connection looks like in the case (the red and black pair are the sensor, and the ribbon cable is the touch interface).

I also renamed the config file for the printer's release year and updated the PR.


![image](https://user-images.githubusercontent.com/13139373/108524902-632bda80-7284-11eb-9bee-942075def1a3.png)

KevinOConnor commented:
Thanks.  I committed this and made a couple of minor changes on top (commit 093db517).

-Kevin
